agent/service.go

```diff
@@ -154,7 +155,25 @@ func (ts *TaskService) Create(requestCtx context.Context, req *taskAPI.CreateTas
 		return nil, fmt.Errorf("Drive %q could not be found", driveID)
 	}
 
-	err = bundleDir.MountRootfs(drive.Path(), "ext4", nil)
+	const (
+		maxRetries = 100
+		retryDelay = 10 * time.Millisecond
+	)
+
+	// We retry here due to Firecracker needing some time to populate the guest
+	// drive.
+	// https://github.com/firecracker-microvm/firecracker/issues/1159
+	for i := 0; i < maxRetries; i++ {
+		if err = bundleDir.MountRootfs(drive.Path(), "ext4", nil); isRetryableMountError(err) {
+			logger.WithError(err).Warnf("failed to mount rootfs %q", drive.Path())
```

Comment:
nit: Including `will retry` or something similar in the error message would allay concerns for someone seeing that in the logs that's not familiar with the background here.

Reply:
Done!